src/app/space/create/deployments/apps/deployments-apps.component.html

```diff
@@ -1,32 +1,82 @@
 <deployments-toolbar (onFilterChange)="filterChange($event)" (onSortChange)="sortChange($event)" [resultsCount]="resultsCount">
 </deployments-toolbar>
 
-<div class="cards-pf">
-  <div class="row row-cards-pf">
-    <div class="col-xs-12 col-sm-4 col-md-2">
-      <h3>
-        <b>Applications</b>
-      </h3>
-    </div>
-    <div class="col-xs-12 col-sm-4 col-md-2">
-      <h3>
-        <b>Environments</b>
-      </h3>
+<div *ngIf="(applications.first() | async); else loading">
+  <div class="cards-pf">
+    <div class="row row-cards-pf">
+      <div class="col-xs-12 col-sm-4 col-md-2">
+        <h3>
+          <b>Applications</b>
+        </h3>
+      </div>
+      <div class="col-xs-12 col-sm-4 col-md-2">
+        <h3>
+          <b>Environments</b>
+        </h3>
+      </div>
     </div>
   </div>
-</div>
 
-<div class="cards-pf">
-  <div class="row row-cards-pf">
-    <div class="col-xs-12 col-sm-4 col-md-2"></div>
-    <div class="col-xs-12 col-sm-4 col-md-3" *ngFor="let environment of environments | async">
-      <h4 class="env-card-title">
-        {{environment.name}}
-      </h4>
+  <div class="cards-pf">
+    <div class="row row-cards-pf">
+      <div class="col-xs-12 col-sm-4 col-md-2"></div>
+      <div class="col-xs-12 col-sm-4 col-md-3" *ngFor="let environment of environments | async">
+        <h4 class="env-card-title">
+          {{environment.name}}
+        </h4>
+      </div>
     </div>
   </div>
-</div>
 
-<div *ngFor="let application of filteredApplicationsList">
-  <deployment-card-container [spaceId]="spaceId | async" [application]="application" [environments]="environments"></deployment-card-container>
+  <div *ngFor="let application of filteredApplicationsList">
+    <deployment-card-container [spaceId]="spaceId | async" [application]="application" [environments]="environments"></deployment-card-container>
+  </div>
 </div>
+
+<ng-template #loading>
+  <div>
+    <div class="cards-pf">
+        <div class="row row-cards-pf">
+          <div>
+            <div class="spinner"></div>
+            <h3 id="loading-msg">Loading Deployments...</h3>
+          </div>
+        </div>
+      </div>
+
+    <div class="cards-pf">
+      <div class="row row-cards-pf">
+        <div class="spacer col-xs-12 col-sm-4 col-md-2">&nbsp;</div>
+        <div class="filler col-xs-12 col-sm-4 col-md-3">&nbsp;</div>
+        <div class="filler col-xs-12 col-sm-4 col-md-3">&nbsp;</div>
+      </div>
+    </div>
```

Comment:
It'd be nice to use `*ngFor` here rather than copy-pasting the placeholder cards.